app/screens/redeem-lnurl-withdrawal-screen/redeem-bitcoin-detail-screen.tsx

```diff
@@ -0,0 +1,445 @@
+  },
+})
+const RedeemBitcoinDetailScreen = ({
+  navigation,
+  route,
+}: StackScreenProps<RootStackParamList, "redeemBitcoinDetail">) => {
+  const { callback, domain, defaultDescription, k1, minWithdrawable, maxWithdrawable } =
+    route.params
+  const minWithdrawableSatoshis = minWithdrawable / 1_000
+  const maxWithdrawableSatoshis = maxWithdrawable / 1_000
+
+  const [receiveCurrency, setReceiveCurrency] = useState<WalletCurrency>(
+    WalletCurrency.Btc,
+  )
+  const { LL } = useI18nContext()
+
+  useEffect(() => {
+    if (receiveCurrency === WalletCurrency.Usd) {
+      navigation.setOptions({ title: LL.RedeemBitcoinScreen.usdTitle() })
+    }
+
+    if (receiveCurrency === WalletCurrency.Btc) {
+      navigation.setOptions({ title: LL.RedeemBitcoinScreen.title() })
+    }
+  }, [receiveCurrency, navigation, LL])
+
+  // useEffect(() => {
+  //   const notifRequest = async () => {
+  //     const waitUntilAuthorizationWindow = 5000
+
+  //     if (Platform.OS === "ios") {
+  //       if (await hasFullPermissions()) {
+  //         return
+  //       }
+
+  //       setTimeout(
+  //         () =>
+  //           Alert.alert(
+  //             LL.common.notification(),
+  //             LL.ReceiveBitcoinScreen.activateNotifications(),
+  //             [
+  //               {
+  //                 text: LL.common.later(),
+  //                 // todo: add analytics
+  //                 onPress: () => console.log("Cancel/Later Pressed"),
+  //                 style: "cancel",
+  //               },
+  //               {
+  //                 text: LL.common.ok(),
+  //                 onPress: () => hasToken && requestPermission(client),
+  //               },
+  //             ],
+  //             { cancelable: true },
+  //           ),
+  //         waitUntilAuthorizationWindow,
+  //       )
+  //     }
+  //   }
+  //   notifRequest()
+  //   // eslint-disable-next-line react-hooks/exhaustive-deps
+  // }, [client, hasToken])
+
+  const [satAmount, setSatAmount] = useState(minWithdrawableSatoshis)
+  const { convertCurrencyAmount } = usePriceConversion()
+  const satAmountInUsd = convertCurrencyAmount({
+    amount: satAmount,
+    from: "BTC",
+    to: "USD",
+  })
+  const minSatAmountInUsd = convertCurrencyAmount({
+    amount: satAmount,
+    from: "BTC",
+    to: "USD",
+  })
+  const maxSatAmountInUsd = convertCurrencyAmount({
+    amount: satAmount,
+    from: "BTC",
+    to: "USD",
+  })
+  const [usdAmount, setUsdAmount] = useState(satAmountInUsd)
+
+  const [amountCurrency, setAmountCurrency] = useState("BTC")
+
+  // const { btcWalletId } = useMainQuery()
+  const { data } = useReceiveBtcQuery({ fetchPolicy: "cache-first" })
+  const btcWalletId = data?.me?.defaultAccount?.btcWallet?.id
+
+  const usdWalletId = null // TODO: when usd wallet ln invoices can be generated providing the sats amount as in put we can have the usdWalletId from useMainQuery as follows: const { usdWalletId } = useMainQuery()
```

Comment:
@daviroo  can this be done using price val?

Reply:
We could try and create an invoice that was the correct sat amount using the usd wallet, but we can only specify usd amounts so we potentially could never get the correct amount of sats.